shell/src/main/java/org/apache/accumulo/shell/Shell.java

```diff
@@ -543,23 +551,17 @@ public int start() throws IOException {
     if (!accumuloDir.exists() && !accumuloDir.mkdirs()) {
       log.warn("Unable to make directory for history at {}", accumuloDir);
     }
-    try {
-      final FileHistory history = new FileHistory(new File(historyPath));
-      reader.setHistory(history);
-      // Add shutdown hook to flush file history, per jline javadocs
-      Runtime.getRuntime().addShutdownHook(new Thread(() -> {
-        try {
-          history.flush();
-        } catch (IOException e) {
-          log.warn("Could not flush history to file.");
-        }
-      }));
-    } catch (IOException e) {
-      log.warn("Unable to load history file at {}", historyPath);
-    }
+
+    // Remove Timestamps for history file. Fixes incompatibility issues
+    reader.unsetOpt(LineReader.Option.HISTORY_TIMESTAMPED);
+
+    // Set history file
+    reader.setVariable(LineReader.HISTORY_FILE, new File(historyPath));
 
     // Turn Ctrl+C into Exception instead of JVM exit
-    reader.setHandleUserInterrupt(true);
+    // Not 100% sure this is necessary anymore.
```

Comment:
Did you test CTRL-C without the lines below?

Reply:
I believe CTRL-C still worked without it but I will double check tomorrow to verify.

Reply:
In my testing, both the main branch and these changes do have the same functionality for CTRL-C but they both might be wrong. When I  hit CTRL-C, it exits the shell which I do not think is what we are going for. I tested the jline3 demo and CTRL-C will exit back to the shell but not exit the shell. That is probably the functionality we want. The funny thing is, I am implementing what they did in their demo but for some reason, it does not seem to be working. So this will take some more investigating.

Reply:
I did find that when I CTRL-C in the middle of the help command, it does work properly. As in it throws the interrupt exception but doesn't exit the process. It only exits if I CTRL-C when just sitting in the shell.

Reply:
This is because of the block below. Not sure if this is a feature that we want or not.

```java
catch (UserInterruptException uie) {
        // User Cancelled (Ctrl+C)
        reader.println();
        writer.println();

        String partialLine = uie.getPartialLine();
        if (partialLine == null || "".equals(uie.getPartialLine().trim())) {
          // No content, actually exit
          return exitCode;
        }
```